src/tools/rbd/action/Kernel.cc

```diff
@@ -153,6 +153,39 @@ static int do_kernel_showmapped(Formatter *f)
   return r;
 }
 
+static void print_error_description(const char *poolname, const char *imgname,
+				    const char *snapname, int maperrno)
+{
+  int r;
+  uint8_t oldformat;
+  librados::Rados rados;
+  librados::IoCtx ioctx;
+  librbd::Image image;
+
+  if (maperrno == ENOENT)
+    return;
+
+  r = utils::init_and_open_image(poolname, imgname, snapname,
+				 true, &rados, &ioctx, &image);
+  if (r < 0)
+    return;
+
+  r = image.old_format(&oldformat);
+  if (r < 0)
+    return;
+
+  std::cout << "check dmesg(1) for rbd related kernel messages.";
```

Comment:
Make it "In some cases useful info is found in syslog - try dmesg | tail or so." and print it last, after all other suggestions, so it looks like:

rbd: sysfs write failed
rbd: map failed: ...
suggestion 1
suggestion 2
there may be something in the dmesg

There may be more than one suggestion we'd like to make in the future, and only a subset of errors is annotated by krbd.